src/WooCommerceSubscriptionsController.php

```diff
@@ -159,4 +164,65 @@ public function maybe_dont_update_payment_method( $update, $new_payment_method,
 
 		return $update;
 	}
+
+	/**
+	 * Status update.
+	 *
+	 * @param Payment $payment Payment.
+	 * @return void
+	 */
+	public static function status_update( Payment $payment ) {
+		$source_id = $payment->get_source_id();
+
+		/**
+		 * Retrieve WooCommerce order from payment source ID,
+		 * if no order is found return early.
+		 *
+		 * @link https://docs.woocommerce.com/wc-apidocs/function-wc_get_order.html
+		 */
+		$order = \wc_get_order( $source_id );
+
+		if ( false === $order ) {
+			return;
+		}
+
+		/**
+		 * This status update function will not update WooCommerce subscription orders.
+		 * 
+		 * @link https://github.com/pronamic/wp-pronamic-pay-woocommerce/issues/48
+		 */
+		if ( 'shop_subscription' !== $order->get_type() ) {
+			return;
+		}
+
+		/**
+		 * Cross-check payment ID.
+		 */
+		$order_payment_id = (int) $order->get_meta( '_pronamic_payment_id' );
+
+		if ( $order_payment_id !== $payment->get_id() ) {
+			return;
+		}
+
+		/**
+		 * Status check.
+		 */
+		if ( ! \in_array( $payment->get_status(), [ PaymentStatus::AUTHORIZED, PaymentStatus::SUCCESS ], true ) ) {
+			return;
+		}
+
+		/**
+		 * Payment method check.
+		 */
+		$payment_method = (string) $payment->get_meta( 'woocommerce_payment_method' );
+
+		if ( '' === $payment_method ) {
+			return;
+		}
+
+		/**
+		 * Update order payment method.
+		 */
+		WC_Subscriptions_Change_Payment_Gateway::update_payment_method( $order, $payment_method );
```

Comment:
Check if class exists before using it or handle in a way (maybe by adding a note about not being able to update the subscription payment method) that will not result in a fatal error if the class does not exist anymore?

Reply:
I've considered it, but shouldn't we expect this feature to be available at this point? In other extensions we also do not check the availability of each extension class, function, etc. If WooCommerce Subscriptions removes the method, a (fatal) error may occur. A note with an order, subscription and/or payment is not necessarily noticeable.